x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlSession.java

```diff
@@ -143,12 +150,106 @@ public void execute(
+            executeOptimizedPlan(request, executionInfo, runPhase, optimizedPlan(analyzedPlan), listener);
+        }
+
+        /**
+         * Whether to return an empty result (HTTP status 200) for a CCS rather than a top level 4xx/5xx error.
+         *
+         * For cases where field-caps had no indices to search and the remotes were unavailable, we
+         * return an empty successful response (200) if all remotes are marked with skip_unavailable=true.
+         *
+         * Note: a follow-on PR will expand this logic to handle cases where no indices could be found to match
+         * on any of the requested clusters.
+         */
+        private boolean returnSuccessWithEmptyResult(Exception e) {
+            if (executionInfo.isCrossClusterSearch() == false) {
+                return false;
+            }
+
+            if ((e instanceof IllegalStateException ill && ill.getMessage().equals("No clusters to search"))
+                || ExceptionsHelper.isRemoteUnavailableException(e)) {
+                for (String clusterAlias : executionInfo.clusterAliases()) {
+                    if (executionInfo.isSkipUnavailable(clusterAlias) == false
+                        && clusterAlias.equals(RemoteClusterAware.LOCAL_CLUSTER_GROUP_KEY) == false) {
+                        return false;
+                    }
+                }
+                return true;
+            }
+            return false;
+        }
+
+        @Override
+        public void onFailure(Exception e) {
+            if (returnSuccessWithEmptyResult(e)) {
+                executionInfo.markEndQuery();
+                Exception exceptionForResponse;
+                if (e instanceof ConnectTransportException) {
+                    // when field-caps has no field info (since no clusters could be connected to or had matching indices)
+                    // it just throws the first exception in its list, so this odd special handling is here is to avoid
+                    // having one specific remote alias name in all failure lists in the metadata response
+                    exceptionForResponse = new RemoteTransportException(
+                        "connect_transport_exception - unable to connect to remote cluster",
+                        null
+                    );
+                } else {
+                    exceptionForResponse = e;
+                }
+                for (String clusterAlias : executionInfo.clusterAliases()) {
+                    executionInfo.swapCluster(clusterAlias, (k, v) -> {
+                        EsqlExecutionInfo.Cluster.Builder builder = new EsqlExecutionInfo.Cluster.Builder(v).setTook(
+                            executionInfo.overallTook()
+                        ).setTotalShards(0).setSuccessfulShards(0).setSkippedShards(0).setFailedShards(0);
+                        if (RemoteClusterAware.LOCAL_CLUSTER_GROUP_KEY.equals(clusterAlias)) {
+                            // never mark local cluster as skipped
```

Comment:
Example where this is needed:

```
  "query": "from noindex*,remote1:employees,remote2:noindex*|\n STATS count(*)",
```

where remote1 is down (unavailable) and both remote clusters are `skip_un=true`.
What happens is that since there are no indexes to search on any of the requested clusters and since both remotes were skip_un=true and since the local cluster indices were specified with a wildcard (lenient handling), we don't throw a fatal exception.

But in discussion with Andrei, the local cluster will never be marked as skipped. We may also change this to be a fatal scenario in the follow on ticket where we will deal with "missing indices" failures, so for now the rule is just to not mark local as skipped.

For the above case, without the special handling in this code, you get incorrect information in the "(local)" cluster/details entry: 
```
  "_clusters": {
    "total": 3,
    "successful": 0,
    "running": 0,
    "skipped": 3,
    "partial": 0,
    "failed": 0,
    "details": {
      "(local)": {
        "status": "skipped",
        "indices": "noindex*",
        "took": 18,
        "_shards": {
          "total": 0,
          "successful": 0,
          "skipped": 0,
          "failed": 0
        },
        "failures": [
          {
            "shard": -1,
            "index": null,
            "reason": {
              "type": "remote_transport_exception",
              "reason": "[connect_transport_exception - unable to connect to remote cluster]"
            }
          }
        ]
      },
      "remote2": {
        "status": "skipped",
        "indices": "noindex*",
        "took": 18,
        "_shards": {
          "total": 0,
          "successful": 0,
          "skipped": 0,
          "failed": 0
        },
        "failures": [
          {
            "shard": -1,
            "index": null,
            "reason": {
              "type": "remote_transport_exception",
              "reason": "[connect_transport_exception - unable to connect to remote cluster]"
            }
          }
        ]
      },
      "remote1": {
        "status": "skipped",
        "indices": "employees",
        "took": 18,
        "_shards": {
          "total": 0,
          "successful": 0,
          "skipped": 0,
          "failed": 0
        },
        "failures": [
          {
            "shard": -1,
            "index": null,
            "reason": {
              "type": "remote_transport_exception",
              "reason": "[connect_transport_exception - unable to connect to remote cluster]"
            }
          }
        ]
      }
    }
  }
```

Reply:
I didn't have a test for this case, so I've added one in CrossClusterQueryUnavailableRemotesIT on the next push.